dev/com.ibm.ws.webcontainer.servlet.4.0_fat/fat/src/com/ibm/ws/fat/wc/tests/WCSameContextRootTest.java

```diff
@@ -0,0 +1,107 @@
+import org.apache.hc.client5.http.impl.classic.CloseableHttpResponse;
+import org.apache.hc.client5.http.impl.classic.HttpClientBuilder;
+import org.apache.hc.core5.http.io.entity.EntityUtils;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import com.ibm.websphere.simplicity.ShrinkHelper;
+
+import componenttest.annotation.ExpectedFFDC;
+import componenttest.annotation.Server;
+import componenttest.custom.junit.runner.FATRunner;
+import componenttest.custom.junit.runner.Mode;
+import componenttest.custom.junit.runner.Mode.TestMode;
+import componenttest.topology.impl.LibertyServer;
+
+/**
+ * Starting applications with same context root - verify that the second starting app with same context root can not
+ * stop the first started app. Also the subsequent app can not start with the exception SRVE0164E "context.root.already.in.use"
+ *
+ * Note: server.xml uses the "startAfter" attribute to control the app startup order. Need that in order to have consistent behavior
+ */
+@RunWith(FATRunner.class)
+@Mode(TestMode.FULL)
+public class WCSameContextRootTest {
+    private static final Logger LOG = Logger.getLogger(WCSameContextRootTest.class.getName());
+    private static final String APP1_NAME = "SameContextRootTestApp1";
+    private static final String APP2_NAME = "SameContextRootTestApp2";
+
+    @Server("servlet40_SameContextRootTestServer")
+    public static LibertyServer server;
+
+    @BeforeClass
+    @ExpectedFFDC({ "com.ibm.ws.webcontainer.exception.WebAppNotLoadedException", "com.ibm.ws.container.service.state.StateChangeException" })
+    public static void setUp() throws Exception {
+        LOG.info("Setup 2 application with same context root: " + APP1_NAME + " and " + APP2_NAME);
+
+        //put in /apps folder
+        ShrinkHelper.defaultApp(server, APP1_NAME + ".war", "app1.same.contextroot");
+        ShrinkHelper.defaultApp(server, APP2_NAME + ".war", "app2.same.contextroot");
+
+        // We can't start the server like normal as there are application startup errors.
+        // Instead we'll set the log name, start the server and won't validate the applications.
+        // Then we'll validate the applications on our own.
+        server.setConsoleLogName(WCSameContextRootTest.class.getSimpleName() + ".log");
+        server.startServerAndValidate(true, true, false);
+
+        // Validate the necessary application has started.
+        server.validateAppLoaded(APP1_NAME);
+
+        // wait for the three ffdc
+        server.waitForMultipleStringsInLog(3, "FFDC1015I");
+        LOG.info("Setup : complete, ready for Tests");
+    }
+
+    @AfterClass
+    public static void tearDown() throws Exception {
+        LOG.info("tearDown : stop server");
+
+        if (server != null && server.isStarted()) {
+            //Expecting the Error during the startModule
+            server.stopServer("SRVE0164E:.*", "CWWKZ0002E:.*");
```

Comment:
Can you just put a brief comment about why we're expecting these errors? Perhaps give an example from the logs so that it is clear?